store/sdk/src/main/java/org/apache/carbondata/sdk/file/Field.java

```diff
@@ -78,6 +78,8 @@ public Field(String name, String type) {
       this.type = DataTypes.FLOAT;
     } else if (type.equalsIgnoreCase("double")) {
       this.type = DataTypes.DOUBLE;
+    } else if (type.equalsIgnoreCase("decimal")) {
+      this.type = DataTypes.createDefaultDecimalType();
```

Comment:
Field has another constructor, need to handle there also ?

Reply:
Also creating Default is not a good idea, user might want to set scale and precision. That's why this constructor was not added. 
Refer `testAllPrimitiveDataType` testcase where decimal is called with other Field constructor

Reply:
Schema.parseJson(schema) doesn't support decimal. So I support default decimal in here, we can support if need set scale and precision.

Reply:
But if parse json takes default precision, actual precision may be different, so it will result in data mismatch

Reply:
already support in new commit